### PR TITLE
Added check for changes upon updating proposal

### PR DESCRIPTION
**Pull request by nickbristow**
If no model changes are made, it doesn't hit the DB and just redirects to the proposals show page.

### CODE REVIEW

File: app/controllers/use_case_controller.rb [hunk 1]
```diff
@@ -41,7 +41,9 @@ def edit
   def update
     @model_instance.assign_attributes(self.permitted_params)   # don't hit db yet
 
-    if self.errors.empty?
+    if @model_instance.changed_attributes.blank? && self.errors.empty?
```
Comment (cmc333333) [inline app/controllers/use_case_controller.rb]: Does this still account for a situation where 1) the fields don't change but 2) a different approver is selected?

Unrelated: probably a good idea to add a flash notice

**Comment by nickbristow** [comment]
@cmc333333 I changed how this works slightly. 

It goes through as normal if you change the email, I also added the flash response as noted in @phirefly 's comment.

File: app/controllers/use_case_controller.rb [hunk 2]
```diff
@@ -54,6 +57,19 @@ def update
...
+  end
+
+  def approver_email_changed?
+    email = params[:approver_email]
+    if !email.nil?
+      @model_instance.approval_changed?(email)
+    else
+      false
+    end
+  end
```
Comment (cmc333333) [inline app/controllers/use_case_controller.rb]: I think this logic is specific to NCR, right? How about, in `UseCaseController`:
```ruby
def attribute_changes?
    !@model_instance.changed_attributes.blank?
end
```

Then, in the `Ncr::WorkOrderController`
```ruby
def attribute_changes?
    super || @model_instance.approval_changed?(@approver_email)
end
```

File: app/models/ncr/work_order.rb [hunk 3]
```diff
@@ -54,6 +54,11 @@ def update_approver(approver_email)
       # no need to call initialize_approvals as they have already been set up
     end
 
+    def approval_changed? approval_email
```
Comment (cmc333333) [inline app/models/ncr/work_order.rb]: Style thing: parens are generally included when defining a method

Comment (cmc333333) [inline app/models/ncr/work_order.rb]: Also, suggestion on naming: `def approver_changed?(new_email)`

File: app/controllers/use_case_controller.rb [hunk 4]
```diff
@@ -54,6 +57,19 @@ def update
 
   protected
 
+  def attribute_changes?
+    !@model_instance.changed_attributes.blank?
+  end
+
+  def approver_email_changed?
```
Comment (cmc333333) [inline app/controllers/use_case_controller.rb]: This method's no longer ever called, hey?

File: spec/features/ncr_work_orders_spec.rb [hunk 5]
```diff
@@ -207,6 +207,16 @@
...
     end
 
+    it "does not resave unchanged requests" do 
+      visit "/ncr/work_orders/#{work_order.id}/edit"
+      click_on 'Submit for approval'
+      visit "/ncr/work_orders/#{work_order.id}/edit"
+      click_on 'Submit for approval'
+
+      expect(current_path).to eq("/proposals/#{work_order.proposal.id}")
+      expect(page).to have_content("No changes were made to the request")
```
Comment (cmc333333) [inline spec/features/ncr_work_orders_spec.rb]: Might be good to test that no emails were sent out, either. You should be able to use `deliveries` for this

**Comment by cmc333333** [comment]
Sorry to do it @nickbristow , but would you mind upgrading bundle to fix the `Gemfile.lock` file. Should just be `gem update bundler`, I think

**Comment by cmc333333** [comment]
@nickbristow agreed with pundit tweaks, so merging

**Pull request merged**